website/preprints/tasks.py

```diff
@@ -18,29 +19,60 @@
 
 
 @celery_app.task(ignore_results=True, max_retries=5, default_retry_delay=60)
-def on_preprint_updated(preprint_id, update_share=True, share_type=None, old_subjects=None):
+def on_preprint_updated(preprint_id, update_share=True, share_type=None, old_subjects=None, saved_fields=None):
     # WARNING: Only perform Read-Only operations in an asynchronous task, until Repeatable Read/Serializable
     # transactions are implemented in View and Task application layers.
     from osf.models import PreprintService
     preprint = PreprintService.load(preprint_id)
     if old_subjects is None:
         old_subjects = []
-    if preprint.node:
+    if should_send_preprint_metadata(preprint, old_subjects, saved_fields):
         update_or_create_preprint_identifiers(preprint)
     if update_share:
         update_preprint_share(preprint, old_subjects, share_type)
 
+def should_send_preprint_metadata(preprint, old_subjects, saved_fields):
+    # Only call for a metadata update if identifiers haven't been requested yet, or subjects aren't being set
+    send_preprint_metadata = True
```

Comment:
Drive-by: This function can be simplified to

```python
# Only update identifier metadata iff...
return (
    # the preprint is valid (has a node)
    preprint.node and
    # DOI didn't just get created
    not (saved_fields and 'preprint_doi_created' in saved_fields) and
    # subjects aren't being set
    not old_subjects
)
```

Also, the name is a bit a bit vague: in the context of this module, "preprint metadata" could refer to the metadata sent to either Crossref or SHARE. How about `should_update_preprint_identifiers`?